openpype/hosts/maya/api/plugin.py

```diff
@@ -247,6 +247,11 @@ def update(self, container, representation):
 
             self.log.warning("Ignoring file read error:\n%s", exc)
 
+        shapes = cmds.ls(content, shapes=True, long=True)
+        new_nodes = (list(set(content) - set(shapes)))
```

Comment:
I started to type the comment below - but I noticed you are trying to mimic the behavior that `process_reference` does for the `ReferenceLoader`. I believe there original shape filtering was once done to avoid shapes being added to the containers. Which has since been gone completely anyway - yay!

Anyway, just leaving this comment here as a side note - ignore it for now.

---

I'd recommend to not filter out the shapes. It's redundant.

I believe I can also technically parent (instance) a referenced shape into a transform that I want to currently export. `cmds.parent(referenced_shape, my_transform, add=True, shape=True)` with something like this. So technically I could be extracting a hierarchy that only has referenced shape. We'd still want to collect that reference too.

```my_transform > referenced_shape```

What's the reason you'd want to get these `new_nodes` like this?
You should be able to pass `content` directly to `self._organize_containers`. No?

Reply:
agree, I've just copy pasted the code without thinking much about it.